cachi2/core/paths.py

```diff
@@ -0,0 +1,109 @@
+        created under ``root/temp/``.
+    """
+
+    go_mod_cache_download_part = Path("pkg", "mod", "cache", "download")
+
+    def __new__(cls, root, source_dir, app_subpath=os.curdir):
+        """
+        Create a new Path object.
+
+        :param int request_id: the ID of the request this bundle is for.
+        :param str root: the root directory to the bundles.
+        :param str app_subpath: an optional relative path to where the application resides in the
+            source directory. This sets ``self.source_dir`` and all other related paths to
+            start from that directory. If this is not set, it is assumed the application lives in
+            the root of the source directory.
+        """
+        self = super().__new__(cls, root)
+        self._path_root = root
+
+        self.source_root_dir = Path(source_dir)
+        self.source_dir = self.source_root_dir.joinpath(app_subpath)
+        self.go_mod_file = self.source_dir.joinpath("go.mod")
+
+        self.deps_dir = self.joinpath("deps")
+        self.gomod_download_dir = self.joinpath("deps", "gomod", cls.go_mod_cache_download_part)
+
+        self.node_modules = self.source_dir.joinpath("node_modules")
+        self.npm_deps_dir = self.joinpath("deps", "npm")
+        self.npm_package_file = self.source_dir.joinpath("package.json")
+        self.npm_package_lock_file = self.source_dir.joinpath("package-lock.json")
+        self.npm_shrinkwrap_file = self.source_dir.joinpath("npm-shrinkwrap.json")
+
+        self.pip_deps_dir = self.joinpath("deps", "pip")
+
+        self.rubygems_deps_dir = self.joinpath("deps", "rubygems")
+
+        self.yarn_deps_dir = self.joinpath("deps", "yarn")
+
+        self.bundle_archive_file = Path(root, f"bundle.tar.gz")
+        self.bundle_archive_checksum = Path(root, f"bundle.checksum.sha256")
+
+        self.packages_data = Path(root, f"packages.json")
+        self.gomod_packages_data = self.joinpath("gomod_packages.json")
+        self.npm_packages_data = self.joinpath("npm_packages.json")
+        self.pip_packages_data = self.joinpath("pip_packages.json")
+        self.yarn_packages_data = self.joinpath("yarn_packages.json")
+        self.rubygems_packages_data = self.joinpath("rubygems_packages.json")
+        self.git_submodule_packages_data = self.joinpath("git_submodule_packages.json")
+
+        return self
+
+    def app_subpath(self, subpath):
+        """Create a new ``RequestBundleDir`` object with the sources pointed to the subpath."""
+        return BaseRequestBundleDir(self._path_root, self.source_dir, subpath)
+
+    def relpath(self, path):
+        """Get the relative path of a path from the root of the source directory."""
+        return os.path.relpath(path, start=self.source_root_dir)
+
+    def rmtree(self):
+        """Remove this directory tree entirely."""
+        shutil.rmtree(str(self))
+
+
+class RequestBundleDir(BaseRequestBundleDir):
```

Comment:
Oh yeah, these two need to be shot

And IMO we should just call it OutputDir

Reply:
I did a simple implementation getting rid of both. We might need to have a more complex mechanism later, as more package managers are introduced.